src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,80 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  todos: Todo[];
+  select?: number;
+  setSelect: (select?: number) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todos, select, setSelect }) => {
+  const [loading, setLoading] = useState(false);
+  const [user, setUser] = useState<User>();
+
+  const selectedTodo = todos.find(todo => todo.id === select);
+  const foundedUserId = selectedTodo?.userId;
+
+  useEffect(() => {
+    setLoading(true);
+
+    if (foundedUserId) {
+      getUser(foundedUserId)
+        .then(userFromServer => setUser(userFromServer))
+        .finally(() => setLoading(false));
+    }
```

Comment:
The loading state is set to true at the beginning of the `useEffect`, but it is only set to false if a user is fetched. If `foundedUserId` is undefined, the loading state remains true, which could cause the loader to display indefinitely. Consider setting the loading state to false when `foundedUserId` is undefined.